adapter/internal/oasparser/envoyconf/response_mapper.go

```diff
@@ -75,3 +107,107 @@ func genErrorResponseMapper(statusCode uint32, errorCode int32, message string,
 	}
 	return mapper
 }
+
+func genSoap12ErrorResponseMapper(flag string, statusCode uint32, errorCode int32, message string, description string) *hcmv3.ResponseMapper {
+	msg, _ := soaputils.GenerateSoapFaultMessage(soap12ProtocolVersion, message, description, strconv.Itoa(int(errorCode)))
+
+	mapper := &hcmv3.ResponseMapper{
+		Filter: &access_logv3.AccessLogFilter{
+			FilterSpecifier: &access_logv3.AccessLogFilter_AndFilter{
+				AndFilter: &access_logv3.AndFilter{
+					Filters: []*access_logv3.AccessLogFilter{
+						{
+							FilterSpecifier: &access_logv3.AccessLogFilter_HeaderFilter{
+								HeaderFilter: &access_logv3.HeaderFilter{
```

Comment:
Shall we create the HeaderFilter config via a sub method? The headers we match will be more visible that way.

Reply:
It's bit difficult to take the HeaderFilter into a separate methods, as in come cases its using exact header match and some cases it's a header present match.
Solution that comes to my mind is to have a If statement and check whether its a exact match or present match.
Do we have any other nice way for this? 🤔

Reply:
fixed via https://github.com/wso2/product-microgateway/pull/2972/commits/5c91048f9fa451e4b2f2d91dac704d3f6b755cd3